tests/gradients/core/test_pulse_gradient.py

```diff
@@ -803,8 +803,41 @@ def test_shots_attribute(self, shots):
 
 
 @pytest.mark.jax
-class TestStochPulseGradQNodeIntegration:
-    """Test that stoch_pulse_grad integrates correctly with QNodes."""
+class TestStochPulseGradQNode:
+    """Test that pulse_generator integrates correctly with QNodes."""
+
+    def test_qnode_expval_single_par(self):
+        """Test that a simple qnode that returns an expectation value
+        can be differentiated with pulse_generator."""
+        import jax
+        import jax.numpy as jnp
+
+        jax.config.update("jax_enable_x64", True)
+        dev = qml.device("default.qubit.jax", wires=1)
+        T = 0.2
+        ham_single_q_const = qml.pulse.constant * qml.PauliY(0)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(params):
+            qml.evolve(ham_single_q_const)([params], T)
+            return qml.expval(qml.PauliZ(0))
+
+        params = jnp.array(0.4)
+        with qml.Tracker(dev) as tracker:
+            _match = "stochastic pulse parameter-shift .* scalar pulse parameters."
+            with pytest.warns(UserWarning, match=_match):
+                grad_fn = stoch_pulse_grad(circuit, num_split_times=2)
+            grad = grad_fn(params)
+
+        p = params * T
+        exp_grad = -2 * jnp.sin(2 * p) * T
+        assert jnp.allclose(grad, exp_grad)
+        assert tracker.totals["executions"] == 4  # two shifted tapes, two splitting times
```

Comment:
This is a copy of the test for `pulse_generator`, just to also have at least one test on the `QNode` level for `stoch_pulse_grad`. As it has scalar parameters, it actually works properly. However, as we always raise the warning, we can also check here for the warning being raised.